src/test/regress/sql/qp_dropped_cols.sql

```diff
@@ -8687,3 +8687,112 @@ DELETE FROM dist_key_dropped_pt WHERE b=6;
 -- the tables, or the pg_upgrade test fails.
 set client_min_messages='warning';
 drop schema qp_dropped_cols cascade;
+
+-- Test modifying DML on leaf partition when parent has dropped columns and
+-- the partition has not. Ensure that DML commands pass without execution
+-- errors and produce valid results.
+RESET search_path;
+-- start_ignore
+DROP TABLE IF EXISTS t_part_dropped;
+-- end_ignore
+CREATE TABLE t_part_dropped (c1 int, c2 int, c3 int, c4 int) DISTRIBUTED BY (c1)
+PARTITION BY LIST (c3) (PARTITION p0 VALUES (0));
+
+ALTER TABLE t_part_dropped DROP c2;
+ALTER TABLE t_part_dropped ADD PARTITION p2 VALUES (2);
+
+-- Partition selection should go smoothly when inserting into leaf
+-- partition with different attribute structure.
+INSERT INTO t_part_dropped VALUES (1, 2, 4);
+INSERT INTO t_part_dropped_1_prt_p2 VALUES (1, 2, 4);
+INSERT INTO t_part_dropped_1_prt_p2 VALUES (1, 2, 0);
+
+-- Ensure that split update on leaf and root partitions does not
+-- throw partition selection error in both planners.
+UPDATE t_part_dropped_1_prt_p2 SET c1 = 2;
+UPDATE t_part_dropped SET c1 = 3;
+
+-- Ensure that split update on leaf partition does not throw constraint error
+-- (executor does not choose the wrong partition at insert stage of update).
+INSERT INTO t_part_dropped VALUES (1, 2, 0);
+UPDATE t_part_dropped_1_prt_p2 SET c1 = 2 WHERE c4 = 0;
+
+SELECT count(*) FROM t_part_dropped_1_prt_p2;
+
+-- Split update on root relation should choose the correct partition
+-- at insert (executor doesn't put the tuple to wrong partition for legacy
+-- planner case).
+UPDATE t_part_dropped SET c1 = 3 WHERE c4 = 0;
+SELECT count(*) FROM t_part_dropped_1_prt_p2;
+SELECT * FROM t_part_dropped_1_prt_p0;
+
+-- For ORCA the partition selection error should not occur.
```

Comment:
How can you be sure that the plan was generated by ORCA?

Reply:
Have added explain output for several commands.